### DISCUSSION TITLE
Change value of static field from decorator (2018-09)?

### DISCUSSION THREAD

**Discussion by joezappie** [babel/babel#15429] [Q&A] [answered]
I'm wondering if its possible to change the value of a class variable when using legacy decorators version 2018-09?

```
const addItem = (newValue) => {
  return function (target) {
    target.descriptor.value.push(newValue);
  };
};

class Test {
  @addItem(4)
  static values = [1,2,3];
}

const test = new Test();
console.log(test.constructor.values) // [1,2,3,4]
```

Value on the descriptor is undefined and trying to add it causes the error ` The property descriptor of a field descriptor can't have a .value property.` Is there a different way to accomplish this?

I'm using Lit, and the latest version it supports is unfortunately 2018-09 so no chance of upgrading. My end goal is to make a decorator that can auto add my theme CSS files to the Lit elements static styles array.

I did also try making a class decorator, but since the styles are static, I couldn't figure out a way to get that to work. Ideally I'd love to have a decorator like this, which would auto inject my theming into the styles array.
```
@theme()
class MyComponent extends LitElement {

}
```

**Comment by nicolo-ribaudo** [comment, marked as answer]
This should work:
```js
const addItem = (newValue) => {
  return function (target) {
    const { initializer } = target;
    target.initializer = function () {
      return [
        ...initializer.call(this),
        newValue,
      ];
    };
  };
};

class Test {
  @addItem(4)
  static values = [1,2,3];
}

const test = new Test();
console.log(test.constructor.values) // [1,2,3,4]
```

fields values are wrapped in an `initializer` function.

**Reply by joezappie** [reply]
That worked! Thank you so much.
